snuba/datasets/errors.py

```diff
@@ -1,34 +1,13 @@
 from datetime import timedelta
 from typing import FrozenSet, Mapping, Sequence, Union
 
-from snuba.clickhouse.columns import (
-    Array,
-    ColumnSet,
-    DateTime,
-    FixedString,
-    IPv4,
-    IPv6,
-    LowCardinality,
-    Materialized,
-    Nested,
-    Nullable,
-    String,
-    UInt,
-    UUID,
-    WithCodecs,
-    WithDefault,
-)
 from snuba.datasets.dataset import ColumnSplitSpec, TimeSeriesDataset
-from snuba.datasets.dataset_schemas import StorageSchemas
-from snuba.datasets.errors_processor import ErrorsProcessor
-from snuba.datasets.errors_replacer import ErrorsReplacer, ReplacerState
+from snuba.datasets.errors_replacer import ReplacerState
 from snuba.datasets.plans.single_storage import SingleStorageQueryPlanBuilder
-from snuba.datasets.storage import WritableTableStorage
-from snuba.datasets.schemas.tables import ReplacingMergeTreeSchema
-from snuba.datasets.table_storage import TableWriter, KafkaStreamLoader
 from snuba.datasets.tags_column_processor import TagColumnProcessor
+from snuba.datasets.storages.errors import all_columns, promoted_tag_columns, schema
```

Comment:
The order of the imports is incorrect.

Reply:
So also this is the problem of having a factory. You depend on the factory to get the storage but you actually also depend on the internals of the file that defines the errors storage, which makes it harder to understand why you should use the factory in the first place if you need a storage.

Let's ensure we discuss this point in the afternoon (whether we need a factory at all for #844)

Reply:
Yeah, I hate this. But haven't thought about a solution for it.

Reply:
I think possibly this should be accessed as `storage.get_write_schema()` instead

Reply:
I can't remember whether this was discussed already however ultimately I think promoted tag processor belongs within the storage layer and not the dataset, then this import won't be needed.